cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunTestsTask.java

```diff
@@ -259,17 +263,28 @@ private void generateCoverage(Project project, JBallerinaBackend jBallerinaBacke
             return;
         }
         Map<String, ModuleCoverage> moduleCoverageMap = initializeCoverageMap(project);
+        // Following lists will hold the coverage information needed for the coverage XML file generation.
+        List<ISourceFileCoverage> packageSourceCoverageList = new ArrayList();
+        List<IClassCoverage> packageNativeClassCoverageList = new ArrayList();
+        List<IClassCoverage> packageBalClassCoverageList = new ArrayList();
+        List<ExecutionData> packageExecData = new ArrayList();
+        List<SessionInfo> packageSessionInfo = new ArrayList();
         for (ModuleId moduleId : project.currentPackage().moduleIds()) {
             Module module = project.currentPackage().module(moduleId);
             CoverageReport coverageReport = new CoverageReport(module);
-            coverageReport.generateReport(moduleCoverageMap, jBallerinaBackend, this.includesInCoverage);
+            coverageReport.generateReport(moduleCoverageMap, packageNativeClassCoverageList,
```

Comment:
Shall we add these to `CoverageReport` as constructor parameters? We can create class level variables and pick in the `coverageReport.generateReport` maybe. Ideally we should not have methods that accepts this many arguments. The maximum should be 4, IIRC.

Nonetheless, we can do it as an imporvement. So let's create an issue and proceed with this.

Reply:
Created to track this https://github.com/ballerina-platform/ballerina-lang/issues/29653